.github/workflows/build.yml

```
@@ -0,0 +1,74 @@
+name: terraform
+on:
+  pull_request:
+    branches: [main]
+  push:
+    branches: [main]
+
```

it should cancel in-progress jobs if a new one comes in

```suggestion

concurrency:
  group: terraform-${{ github.event.pull_request.number || github.sha }}
  cancel-in-progress: true

```